### PR TITLE
Bump the version using the #flags from the PR description

### LINKED ISSUE #79: GitHub version bump action always bumps at least minor
Our GitHub version bump action has this version bump combination logic based on commit messages (see also https://github.com/KIT-MRT/util_caching/issues/38#issuecomment-2476138840).
It seems that in our setup with a PR merge commit, this leads to version bumping based on
- the PR commits 
- plus the PR merge commit

In #75, this lead to a minor bump, even though the only PR commit 3586225 contains the `#patch` tag.
That's probably, because the PR merge commit does not contain a bump tag – thus, defaulting to `#minor`

It's a pity we have so much trouble with the version bump action :see_no_evil: 

@ll-nick Any idea how we can fix this? Should we default to `#patch` again?

### PR DESCRIPTION
**Pull request by ll-nick**
Closes #79

Instead of relying on a marketplace action, we now use bash script to compute a new version from the PR description. The new version is computed based on the existence of one of the following flags (proceeded by a #):
- major
- minor
- patch

#patch

### CODE REVIEW

**Comment by ll-nick** [comment]
I tested this using another branch where the workflow is triggered on any push, the version is computed from the commit description. The pipeline also stops after echoing the new version in the next job in order not to generate a bunch of useless releases and automated commits.

Check out the action for a test of each case (major, minor, patch, default)

File: .github/workflows/bump-version-and-create-release-and-push-docker-images.yaml [hunk 1]
```diff
@@ -18,23 +18,13 @@ jobs:
...
-          DEFAULT_BUMP: minor
-          DRY_RUN: true
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          INITIAL_VERSION: ${{ env.INITIAL_VERSION }}
-          WITH_V: true
-
+        run: |
+          # Run the compute_version script and pass the PR description as an argument
+          cd .github/workflows
+          new_tag=$(./compute_version.sh "${{ github.event.pull_request.body }}")
```
Comment (orzechow) [inline .github/workflows/bump-version-and-create-release-and-push-docker-images.yaml:26]: Isn't this vulnerable to code injections?

File: .github/workflows/compute_version.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+set -e
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+VERSION_FILE="$SCRIPT_DIR/../../version"
```
Comment (orzechow) [inline .github/workflows/compute_version.sh:6]: I'd keep the script config-agnostic and pass the file path via argument or env. 
Or even pass the version string to the script directly.

**Review by orzechow — CHANGES_REQUESTED** [review]
Thanks! That was a quick one 👌

Please make sure that this solution is not vulnerable to code injections. Is the PR body escaped or would a PR body starting with e.g. `"` break the job?

File: .github/workflows/bump-version-and-create-release-and-push-docker-images.yaml (hunk 1 above)
Comment (ll-nick) [reply .github/workflows/bump-version-and-create-release-and-push-docker-images.yaml]: I had originally fixed it in [fa95771](https://github.com/KIT-MRT/arbitration_graphs/pull/85/commits/fa957711c76412b5d66c09564187cb057cbd849d) but it turned out that an injection would already break things before the script is called. Therefore I already needed to sanitize the input inside the workflow, see [20e87b8](https://github.com/KIT-MRT/arbitration_graphs/pull/85/commits/20e87b82f896e43ec20d48a7d03bcaa7084420d9). It's now sanitized twice but I guess that doesn't really hurt.

File: .github/workflows/compute_version.sh (hunk 2 above)
Comment (ll-nick) [reply .github/workflows/compute_version.sh]: Fixed in [f08521](https://github.com/KIT-MRT/arbitration_graphs/pull/85/commits/f08521a5601fc0d6bd582e124c3f95227973feab)

**Comment by ll-nick** [comment]
Ok, this is ready for re-review @orzechow 

On top of the points in the discussions above, I tidied up the script and the workflow a little in [95bcfb3](https://github.com/KIT-MRT/arbitration_graphs/pull/85/commits/95bcfb3984ec5a6dc5f1f479b06056cd27060116), [e7934b9](https://github.com/KIT-MRT/arbitration_graphs/pull/85/commits/e7934b95ebc8e9641de81923b273397cddfc22d4) and [228c042](https://github.com/KIT-MRT/arbitration_graphs/pull/85/commits/228c042d837fe5e34db37d876b5371a597d954a3)

**Review by orzechow — APPROVED** [review]
Great! 😁

I guess the last nitpick:
We could think how to avoid the #tag from ending up in our release description. 
Not super important though 🙈

**Comment by ll-nick** [comment]
Thanks for the review. About your point: I'd say the benefit is not worth the effort. It adds another potential point of failure to a workflow that's pretty complex already and I don't think it hurts having it there. So I'll just go ahead and merge as is :)

**Pull request merged**